src/switch_ivr_bridge.c

```diff
@@ -52,24 +52,34 @@ static void text_bridge_thread(switch_core_session_t *session, void *obj)
 	struct vid_helper *vh = obj;
 	switch_status_t status;
 	switch_frame_t *read_frame = 0;
-	switch_channel_t *channel = switch_core_session_get_channel(vh->session_a);
-	switch_channel_t *b_channel = switch_core_session_get_channel(vh->session_b);
+	switch_channel_t *channel;
+	switch_channel_t *b_channel;
 	switch_buffer_t *text_buffer = NULL;
 	switch_size_t text_framesize = 1024, inuse = 0;
 	unsigned char *text_framedata = NULL;
 	switch_frame_t frame = { 0 };
+	switch_core_session_t * session_a, *session_b;
+
+	if ((session_a = switch_core_session_locate(vh->session_a_uuid))) {
+		channel = switch_core_session_get_channel(session_a);
+	}
+
+	if ((session_b = switch_core_session_locate(vh->session_b_uuid))) {
+		b_channel = switch_core_session_get_channel(session_b);
+	}
 
 	switch_buffer_create_dynamic(&text_buffer, 512, 1024, 0);
 	switch_zmalloc(text_framedata, 1024);
 	text_framesize = 1024;
 
 	vh->up = 1;
 
+
 	while (switch_channel_up_nosig(channel) && switch_channel_up_nosig(b_channel) && vh->up == 1) {
-		status = switch_core_session_read_text_frame(vh->session_a, &read_frame, SWITCH_IO_FLAG_NONE, 0);
+		status = switch_core_session_read_text_frame(session_a, &read_frame, SWITCH_IO_FLAG_NONE, 0);
```

Comment:
`session_a` might be a NULL pointer here

Reply:
it seems that we have another assert in `switch_core_session_read_text_frame`. Now, the question is: why do we assert instead of just returning the `SWITCH_STATUS_FALSE` status code here?!